src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -450,6 +451,52 @@ worker_apply_inter_shard_ddl_command(PG_FUNCTION_ARGS)
 
 
 /*
+ * worker_apply_sequence_command takes a CREATE SEQUENCE command string, runs the
+ * CREATE SEQUENCE command then creates and runs an ALTER SEQUENCE statement
+ * which adjusts the minvalue and maxvalue of the sequence such that the sequence
+ * creates globally unique values.
+ */
+Datum
+worker_apply_sequence_command(PG_FUNCTION_ARGS)
+{
+	text *commandText = PG_GETARG_TEXT_P(0);
+	const char *commandString = text_to_cstring(commandText);
+	const char *dummyString = "-";
+	Node *commandNode = ParseTreeNode(commandString);
+	CreateSeqStmt *createSequenceStatement = NULL;
+	AlterSeqStmt *alterSequenceStatement = NULL;
+	char *sequenceName = NULL;
+	char *sequenceSchema = NULL;
+
+	NodeTag nodeType = nodeTag(commandNode);
+	if (nodeType != T_CreateSeqStmt)
+	{
+		ereport(ERROR,
+				(errmsg("must call worker_apply_sequence_command with a CREATE"
+						" SEQUENCE command string")));
+	}
+
+	createSequenceStatement = (CreateSeqStmt *) commandNode;
+	sequenceName = createSequenceStatement->sequence->relname;
+	sequenceSchema = createSequenceStatement->sequence->schemaname;
+
+	/* run the CREATE SEQUENCE command */
+	ProcessUtility(commandNode, commandString, PROCESS_UTILITY_TOPLEVEL,
```

Comment:
Just for my understanding, we wouldn't normally create a sequence here right? Because we create shard placements (which include the sequence) before we get here.